sourcegraph/cody
# Agent: fix Windows issues

Previously, autocomplete wasn't working on Windows for the JetBrains plugin. The changes in this PR fix the problem. I manually verified the fix by running a Windows instance on GCP and setting up IntelliJ via Remote Desktop.

To summarize the diff:

- `vscode.ExtensionContext.extensionPath` was previously an empty object `{}` instead of a `string`.
- `vscode.ExtensionContext.extensionUri` was previously an empty object `{}` instead of a `vscode.Uri`.
- `vscode.Uri.fsPath` didn't work on Windows because it used the URI `path`. This PR fixes the problem by inlining `vscode.Uri` implementation from microsoft/vscode under a new MIT-licensed directory.


## Test plan

- Make sure `src login` is configured
  - `VITEST_ONLY=NotConfigured pnpm run test agent/src/index.test.ts` to verify agent tests are still passing
- `./gradlew clean buildPlugin` in JetBrains plugin
  - Install plugin on Windows computer
  - Confirm completions work as expected
